### DISCUSSION TITLE
Draggable Scroll content?

### DISCUSSION THREAD

**Discussion by najak3d** [AvaloniaUI/Avalonia#6875] [Q&A] [answered]
I am brand new here.   Just started testing with Avalonia tonight, and am very excited with the community and what I've seen so far -- except for ONE MAJOR ISSUE.

I'm very used to Draggable View content for mobile apps.   For example, see how it works for PlayStore content on Android phones.  You use your finger to move the content up/down (and not a scroll bar).

I was shocked to see that Avalonia ScrollViewer does not even permit you to manually control the Vert/Horiz Scroll values yourself.   The only thing it allows is "ScrollToEnd" and "ScrollToHome"...  but nothing in between!   (I was trying to augment these controls by detecting PointerPressed/Released/Moved, and then just force this ScrollViewer to behave as a I wanted.)

I was looking for a way to set "ScrollView.VerticalScrollValue = {some value}" or something like that.

How are we supposed to have scrollable content and allow the user to drag it.    Note, for MANY contexts, this is the expected behavior on mobile (touch screen) interfaces.   But so far, not seeing how Avalonia supports this at all.

Screenshot of PlayStore on Phone, included for clarity:

![image](https://user-images.githubusercontent.com/10914515/140637678-37a08c7d-c46a-4e2e-9a32-fa1acd3adeba.png)

EDIT:  Have just realized that Avalonia was written for "Desktop" not tablets/phones.   This is bad news for me, as we write apps that deploy to ALL devices -- desktops, tablets, and phones.  We're trying to find a better alternative to Xamarin.Forms for apps that run on all platforms (desktop + tablets + phones;  Mac/Windows/iOS/Android).

**Comment by kekekeks** [comment, marked as answer]
Note that touch scrolling is supported by ScrollViewer for touch input devices. You can copy the source code of ScrollGestureRecognizer and remove the [pointer type check](https://github.com/AvaloniaUI/Avalonia/blob/8edfdf6b4978e55ba162d0e424d9527c60a05d9d/src/Avalonia.Input/GestureRecognizers/ScrollGestureRecognizer.cs#L69).
Then you can override ScrollViewer's control template and insert your own recognizer [instead of default one](https://github.com/AvaloniaUI/Avalonia/blob/7842883961d094e08e9def7f30cf32fd573179c7/src/Avalonia.Themes.Fluent/Controls/ScrollViewer.xaml#L39).

**Comment by najak3d** [comment]
I may have just answered my own question, which seems to be that I need to create a derived class of ScrollViewer, which then is able to write to the 'protected' methods such as: 

		protected double VerticalScrollBarValue { get; set; }

So this seems doable.
